pkg/commands/git_commands/commit_loader.go

```diff
@@ -364,23 +358,42 @@ func (self *CommitLoader) setCommitMergedStatuses(refName string, commits []*mod
 			commits[i].Status = models.StatusMerged
 		}
 	}
-	return commits, nil
+	return commits
 }
 
-func (self *CommitLoader) getMergeBase(refName string) (string, error) {
-	info, err := self.getCurrentBranchInfo()
-	if err != nil {
-		return "", err
+func (self *CommitLoader) getMergeBase(refName string) string {
+	if self.quotedBaseBranches == nil {
+		self.quotedBaseBranches = self.getExistingBaseBranches()
 	}
 
-	baseBranch := "master"
-	if strings.HasPrefix(info.RefName, "feature/") {
-		baseBranch = "develop"
+	if *self.quotedBaseBranches == "" {
+		return ""
 	}
 
-	// swallowing error because it's not a big deal; probably because there are no commits yet
-	output, _ := self.cmd.New(fmt.Sprintf("git merge-base %s %s", self.cmd.Quote(refName), self.cmd.Quote(baseBranch))).DontLog().RunWithOutput()
-	return ignoringWarnings(output), nil
+	// We pass all configured base branches to the merge-base call; git will
+	// return the base commit for the closest one.
+	output, err := self.cmd.New(fmt.Sprintf("git merge-base %s %s",
+		self.cmd.Quote(refName), *self.quotedBaseBranches)).DontLog().RunWithOutput()
+	if err != nil {
+		// If there's an error, it must be because one of the base branches that
+		// used to exist when we called getExistingBaseBranches() was deleted
+		// meanwhile. To fix this for next time, throw away our cache.
+		self.quotedBaseBranches = nil
```

Comment:
I only throw away the cache, but don't bother to rerun the command. I guess this situation is rare enough that that's ok.